Apps/phironnet/ironnet_connector.py

```diff
@@ -0,0 +1,371 @@
+
+    def _handle_test_connectivity(self, param):
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        self.save_progress("Attempting to connect to IronAPI")
+
+        # make rest call
+        ret_val, response = self._make_post('/Login', action_result, data=None, headers=None)
+
+        if phantom.is_success(ret_val):
+            self.save_progress("Test Connectivity to IronAPI Passed")
+            return action_result.set_status(phantom.APP_SUCCESS)
+        else:
+            self.save_progress("Test Connectivity to IronAPI Failed")
+            return action_result.set_status(phantom.APP_ERROR)
+
+    def _handle_irondefense_rate_alert(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+
+        # Add an action result object to self (BaseConnector) to represent the action for this param
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        self.save_progress("Received param: {0}".format(param))
+
+        # Access action parameters passed in the 'param' dictionary
+        request = {
+            'alert_id': param['alert_id'],
+            'comment': param['comment'],
+            'share_comment_with_irondome': param['share_comment_with_irondome'],
+            'analyst_severity': severity_mapping[param.get('analyst_severity', '')],
+            'analyst_expectation': expectation_mapping[param.get('analyst_expectation', '')]
+        }
+
+        # make rest call
+        ret_val, response = self._make_post('/RateAlert', action_result, data=request, headers=None)
+
+        # Add the response into the data section
+        action_result.add_data(response)
+
+        if phantom.is_success(ret_val):
+            self.save_progress("Alert rating was successful")
+            return action_result.set_status(phantom.APP_SUCCESS)
+        else:
+            self.save_progress("Alert rating failed")
+            return action_result.set_status(phantom.APP_ERROR)
+
+    def _handle_irondefense_set_alert_status(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+
+        # Add an action result object to self (BaseConnector) to represent the action for this param
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        self.save_progress("Received param: {0}".format(param))
+
+        # Access action parameters passed in the 'param' dictionary
+        request = {
+            'alert_id': param['alert_id'],
+            'comment': param['comment'],
+            'share_comment_with_irondome': param['share_comment_with_irondome'],
+            'status': status_mapping[param['alert_status']]
+        }
+
+        # make rest call
+        ret_val, response = self._make_post('/SetAlertStatus', action_result, data=request, headers=None)
+
+        # Add the response into the data section
+        action_result.add_data(response)
```

Comment:
`action_result` holds the `output` data in JSON

Reply:
Just getting back to looking at these comments now.  Would you like this to change?

Reply:
That is correct to be doing. To add the result data to the JSON file, you should run `/opt/phantom/bin/create_output.py` for each action to get more data in the JSON `output`

Reply:
Thanks - I was able to use that script as suggested and have filled in the output sections.